### PR TITLE
`Exam mode`: Show only exam relevant notifications in the exam mode

### PR DESCRIPTION
**Pull request by coolchock**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I documented the TypeScript code using JSDoc style.

### Motivation and Context

Notifications which are not exam relevant still lead to popups in the bottom right corner when student is in exam mode. This is annoying during an exam and only exam relevant notifications should be displayed on the exam page. Currently there's only one exam relevant notification, which informs about the changes in the problem statement.

### Description

I added a check to determine if the student is currently on the exam mode page. If they are, notifications that are not relevant are not displayed.

#### Exam Mode Testing

Prerequisites:
- 1 Instructor
- 1 Students
- 1 Exam

1. Log in to Artemis
2. Participate in the exam as a student
3. In the second browser/incognito tab log in as an instructor
4. As instructor, create a message (for example announcement) in the course, that the exam is being conducted for, and check if student doesn't receive the course related notification in the bottom right corner
5. As instructor, make changes to the problem statement of an exam exercise. Make sure, that the exercise that you've edit is a part of the student's exam. Please add 'Notification about changes' at the bottom of the edit page. Only this way the notification will be displayed in the exam mode. Check if student received the exam relevant notification.
6. As a student, go to the course page. Then, as instructor create a message in the course again and check if student receives this notification. 

#### Performance Review
- [x] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [ ] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [x] Test 1
- [ ] Test 2
#### Exam Mode Test
- [x] Test 1
- [ ] Test 2

## Summary by CodeRabbit

- **New Features**
	- Enhanced notification system to differentiate and handle notifications specifically in exam mode.
- **Tests**
	- Added and modified tests to ensure accurate notification behavior depending on the user's mode.

### CODE REVIEW

File: src/test/javascript/spec/component/shared/notification/notification-popup.component.spec.ts [hunk 1]
```diff
@@ -198,12 +198,20 @@ describe('Notification Popup Component', () => {
...
             notificationPopupComponent.ngOnInit();
             expect(notificationPopupComponent.notifications).not.toBeEmpty();
         });
 
+        it('should not add received not exam exercise relevant update notification if user is not in exam mode', () => {
+            jest.spyOn(router, 'url', 'get').mockReturnValue('/courses/1/exams/95');
+            replaceSubscribeToNotificationUpdatesUsingQuizNotification();
+            notificationPopupComponent.ngOnInit();
+            expect(notificationPopupComponent.notifications).toBeEmpty();
+        });
```
[the remark this answers is not in the document]
Comment (coolchock) [reply src/test/javascript/spec/component/shared/notification/notification-popup.component.spec.ts:213]: this case is already covered by the existing test cases

**Review by pzdr7 — APPROVED** [review]
Tested on TS2.

- Messages > Announcements and direct messages did not show up during the exam
- Comments on exercises and replies also did not show up
- Exercise change notifications still show up as expected during the exam
- Go back to course as student > Messages are now received normally


-> Works as expected!

**Review by b-fein — APPROVED** [review]
Code changes make sense.

**Review by egekurt123 — APPROVED** [review]
Manually tested on ts6, works as expected

**Review by Predixx — APPROVED** [review]
tested on ts6. Lgtm!

**Pull request merged**